### ISSUE TITLE
Next 13 (and canary) Bundles react and react-dom twice in production build

### ISSUE THREAD

**Issue by Fredkiss3** [vercel/next.js#41814]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:30 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8103
Binaries:
  Node: 16.17.1
  npm: 8.15.0
  Yarn: 1.22.18
  pnpm: 7.1.0
Relevant packages:
  next: 12.3.2-canary.43
  eslint-config-next: 13.0.0
  react: 18.2.0
  react-dom: 18.2.0

### What browser are you using? (if relevant)

Chrome 106.0.5249.119 (arm64)

### Describe the Bug

starting with two simple nextjs project (the getting started template page), one with NextjS 12 and the other with NextjS 13. When I launch a production build, i see that there is almost no difference in the bundle size, even worse i see that with next 12 the bundle is lighter.

Screenshot for next 12 : 
![Capture d’écran 2022-10-25 à 23 42 51](https://user-images.githubusercontent.com/38298743/197893448-2988accc-d9ed-480b-9160-16e43886fa74.png)

Screenshot for next 13 : 

![Capture d’écran 2022-10-25 à 23 42 30](https://user-images.githubusercontent.com/38298743/197893516-d2eaecb4-5e28-4dfb-b261-8a64d4eb6743.png)

When I tried to see the content of the bundle with `@next/bundle-analyzer`, I noticed that nextjs 13 bundles react and react-dom twice while in nextjs 12 it bundles it only once. 

Screenshot for next 12 : 
![Capture d’écran 2022-10-25 à 23 43 21](https://user-images.githubusercontent.com/38298743/197893566-00cd7376-7fe1-4e39-9092-1d35f7dcd00a.png)

Screenshot for next 13 : 
![Capture d’écran 2022-10-25 à 23 43 11](https://user-images.githubusercontent.com/38298743/197893568-b965ad53-66ed-4882-9c78-88d0a190ebc5.png)

### Expected Behavior

I don't know if it's a bug or the expected behavior, but in my opinion it should not bundle react-dom twice since for an app this size react-dom is the biggest factor in size.

And if it the expected behavior, then i would like to know why it does that.

### Link to reproduction

https://stackblitz.com/edit/vercel-next-js-1wu76s

### To Reproduce

To reproduce, clone the template repo and run a production build with `ANALYZE=true` env variable to see the content of the generated bundle.

**Comment by Chastrlove** [comment]
I'm having the same problem.

![image](https://user-images.githubusercontent.com/17813559/197920753-5738116a-b61c-4e9d-bf49-b17d001923f8.png)

![image](https://user-images.githubusercontent.com/17813559/197920776-a6b1d664-c6ad-4236-a1ad-00f93c240881.png)

**Comment by xiaoyu-tamu** [comment]
same here 🤫

**Comment by antonybudianto** [comment]
Same here, looks like happened on RSC/ new "app" folder routing..

**Comment by balazsorban44** [comment]
Duplicate of #41870. A fix has been merged and will be available soon in the next `canary` release.
